File: AddressBook/react-client/src/dtos/ContactWebsite.ts
import Common from "../utility/Common";
import { PagedReq } from "./PagedReq";
import { ContactRes } from "./Contact";
import { WebsiteLabelRes } from "./WebsiteLabel";

export interface ContactWebsiteRes {
  contactWebsiteId?: string;
  website?: string;
  contactId?: string;
  contact?: ContactRes;
  websiteLabelId?: string;
  websiteLabel?: WebsiteLabelRes;
}

export class ContactWebsiteReqEdit {
  website?: string = "";
  contactId?: string = "";
  websiteLabelId?: string = "";
  constructor(contactId?: string) {
    this.contactId = contactId;
  }
}

export class ContactWebsiteReqSearch extends PagedReq {
  contactId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {contactId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.contactId = contactId;
  }
}

File: AddressBook/Dtos/Dtos/ContactWebsite.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class ContactWebsiteRes
    {
        public int ContactWebsiteId { get; set; }
        public string? Website { get; set; }
        public int? ContactId { get; set; }
        public ContactRes? Contact { get; set; }
        public int? WebsiteLabelId { get; set; }
        public WebsiteLabelRes? WebsiteLabel { get; set; }
    }

    public class ContactWebsiteReqEdit
    {
        [Required]
        public string? Website { get; set; }
        public int? ContactId { get; set; }
        public int? WebsiteLabelId { get; set; }
    }

    public class ContactWebsiteReqSearch : PagedReq
    {
        [Required]
        public int? ContactId { get; set; }
    }
}
